src/utilities.jl

```diff
@@ -145,3 +145,73 @@ end
 
 check_positive_definite(M::AbstractMatrix; kwargs...) =
   check_positive_definite(LinearOperator(M); kwargs...)
+
+
+  """
+  solve_shifted_system!(B, 
+                        x, 
+                        σ, 
+                        γ_inv, 
+                        inv_Cx, 
```

Comment:
Please rename this. This name doesn't suggest anything. Also, you no longer need to pass the arguments that are preallocated inside B.